spec/requests/survey_spec.rb

```diff
@@ -1,7 +1,75 @@
 require 'rails_helper'
 
-RSpec.describe 'Surveys', type: :request do
-  describe 'GET /admin/surveys' do
-    pending "add some examples (or delete) #{__FILE__}"
+RSpec.describe 'Surveys' do
+  describe 'GET admin/surveys#index' do
+    it 'renders the index for user role admin' do
+      sign_in FactoryBot.create(:user, :admin)
+
+      get '/admin/surveys'
+      assert_response :success
+    end
+
+    it 'redirects to home page for users with role user' do
+      sign_in FactoryBot.create(:user, :user)
+
+      get '/admin/surveys'
+      assert_response :redirect
+      follow_redirect!
+      assert_response :success
+      expect(response).to render_template('posts/index')
+    end
+  end
+
+  describe 'GET admin/surveys#new' do
+    it 'should render New Survey template' do
+      sign_in FactoryBot.create(:user, :admin)
+
+      get '/admin/surveys/new'
+      assert_response :success
+      expect(response).to render_template('surveys/new')
+    end
+
+    it 'should redirect to home page' do
+      sign_in FactoryBot.create(:user, :user)
+
+      get '/admin/surveys/new'
+      assert_response :redirect
+      follow_redirect!
+      assert_response :success
+      expect(response).to render_template('posts/index')
+    end
+  end
+
+  describe 'POST /admin/surveys#create' do
+    it 'creates survey with two answers' do
+      sign_in FactoryBot.create(:user, :admin)
+
+      expect do
+        post '/admin/surveys',
+             params: {
+               survey: FactoryBot.attributes_for(:survey).merge(survey_answers_attributes: {
+                                                                  1 => FactoryBot.attributes_for(:survey_answer),
+                                                                  2 => FactoryBot.attributes_for(:survey_answer)
+                                                                })
+             }
+      end.to change { Survey.all.size }.by(1)
+    end
+  end
+
+  describe 'GET /admin/surveys#edit' do
+    it 'should render Edit Survey template' do
+      user = FactoryBot.create(:user, :admin)
```

Comment:
it could be
```ruby
RSpec.describe 'Surveys' do
   let(:admin) { FactoryBot.create(:user, :admin) }
   ....
```
and then `admin` reused in all places where needed

Reply:
done, updated survey tests